odoo/http.py

```diff
@@ -318,7 +318,7 @@ def _handle_exception(self, exception):
 
     def _call_function(self, *args, **kwargs):
         request = self
-        if self.endpoint.routing['type'] != self._request_type:
+        if self.endpoint.routing['type'] != self._request_type and self.endpoint.routing['type'] != 'raw':
```

Comment:
Several comments on this:

- Why not making "type" accept an array instead then?
  So you could do `type=['json', 'http']`.
  "raw" is pretty meaningless and if we ever add a third type of controller method, then it's broken.
- The feature should correctly handle method parameters whether it's called in 'http' or 'json', like other endpoint types.
- If you are going to add such a generic feature, you will need to write a few tests for it.
  (Check that all types work, parameters are correctly parsed, ...)

I understand you need to be able to handle both types, but this commit feels more like a "dirty fix" than a fully working new request type.

Reply:
@awa-odoo It can not be an array, because the "http handler" is a class, and is chosen thanks to the routing type...
So, when the type is `raw`, we take the HTTP handler but we have no restriction on the "request" type

You think I should add a new handler and not using the basic HTTP handler ?

Reply:
@awa-odoo 
Using a list instead of a new parameter might need more code (because all the actual code use a string and not a list), and I prefer to write less code as possible in `http.py` (to avoid to break existing stuff).

Actually, the handler (the object used to parse the HTTP headers, and to call controllers) chose automatically the right handler (JSON or HTTP). And, they check if the handler is correct, regarding the routing type.
The `raw` type just ignore this verification. I think it's the simpler way to do it.

I can think about a better way to do it, but I don't know how

But I agree on the point that I should add test about it.

Reply:
@awa-odoo 
But in the spec, est wanted to have the setting for all media, I'll ask her want she prefer